test/e2e/gc_e2e_test.go

```diff
@@ -284,4 +284,285 @@ var _ = Describe("Garbage collection for dependent resources", func() {
+					Kind:       v1alpha1.CatalogSourceKind,
+					APIVersion: v1alpha1.CatalogSourceCRDAPIVersion,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      sourceName,
+					Namespace: testNamespace,
+					Labels:    map[string]string{"olm.catalogSource": sourceName},
+				},
+				Spec: v1alpha1.CatalogSourceSpec{
+					SourceType: v1alpha1.SourceTypeGrpc,
+					Image:      imageName,
+				},
+			}
+
+			source, err := operatorClient.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.TODO(), source, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred(), "could not create catalog source")
+
+			// Create a Subscription for package
+			_ = createSubscriptionForCatalog(operatorClient, source.GetNamespace(), subName, source.GetName(), packageName, channelName, "", v1alpha1.ApprovalAutomatic)
+
+			// Wait for the Subscription to succeed
+			Eventually(func() error {
+				_, err = fetchSubscription(operatorClient, testNamespace, subName, subscriptionStateAtLatestChecker)
+				return err
+			}).Should(BeNil())
+
+			Eventually(func() error {
+				_, err := kubeClient.GetConfigMap(testNamespace, configmapName)
+				return err
+			}).Should(Succeed(), "expected no error getting configmap object associated with CSV")
+		})
+
+		When("the subscription is updated to a later CSV with a configmap with the same name but new data", func() {
+			const (
+				upgradeChannelName = "beta"
+				newCSVname         = "busybox.v3.0.0"
+			)
+
+			BeforeEach(func() {
+				// update subscription first
+				sub, err := operatorClient.OperatorsV1alpha1().Subscriptions(testNamespace).Get(context.TODO(), subName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred(), "could not get subscription")
+
+				// update channel on sub
+				sub.Spec.Channel = upgradeChannelName
+				_, err = operatorClient.OperatorsV1alpha1().Subscriptions(testNamespace).Update(context.TODO(), sub, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred(), "could not update subscription")
+
+				// Wait for the Subscription to succeed
+				Eventually(func() error {
+					_, err = fetchSubscription(operatorClient, testNamespace, subName, subscriptionStateAtLatestChecker)
+					return err
+				}).Should(BeNil())
+
+				// Ensure the new csv is installed
+				Eventually(func() error {
+					_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), newCSVname, metav1.GetOptions{})
+					return err
+				}).Should(BeNil())
+			})
+
+			It("OLM should have upgraded associated configmap in place", func() {
+				Eventually(func() bool {
+					cfg, err := kubeClient.GetConfigMap(testNamespace, configmapName)
+					if err != nil {
+						return false
+					}
+					// check data in configmap to ensure it is the new data (configmap was updated in the newer bundle)
+					// new value in the configmap is "updated-very-much"
+					data := cfg.Data["special.how"]
+					return data == "updated-very-much"
+				}).Should(BeTrue())
```

Comment:
I think both Expect and Eventually/Consistently can handle multiple return values and implicitly fail if anything after the first return value is non-nil, which would let you write this:

```suggestion
				Eventually(func() (string, error) {
					cfg, err := kubeClient.GetConfigMap(testNamespace, configmapName)
					if err != nil {
						return "", err
					}
					return cfg.Data["special.how"], nil
				}).Should(Equal("updated-very-much"))
```

Reply:
nice